torchao/quantization/qat/linear.py

```diff
@@ -30,6 +30,82 @@
     _get_qmin_qmax,
 )
 
+class SelfCompressionQATQuantizer(torch.nn.Module):
```

Comment:
This feels like a pretty experimental feature. I think this belongs better to a separate file or under torchao/prototype